mzuccaroli/angular-google-tag-manager
# chore: update to Angular 15

Hello,

This PR does:

- Update to latest Angular 14 packages
- Update all packages to latest version except:
  - `typescript`
  - `@types/node`
- Update the lib to only support Angular `^15`
- Update the app to Angular 15
  - `browserlistrc` was removed
  - `environment` files have been deleted
    - `angular.json` was update to reflect that
- Explicit `tslib` version in the lib
- I removed `angular-google-tag-manager` from the installed packages, it now use the one built locally
  - It prevented from updating to Angular 15 (I don't like `--force`)
  - The lib should be built before launching the demo app
  - I added a `paths` entry in the `tsconfig.json` file
- Only update patch packages version
- I reworked the project structure to only have one `package.json`
  - The demo-app is now in the folder `projects`
  - `angular.json` has been updated
  - This allowed to use the locally built lib on the demo-app (else there was a problem with Angular 15)

Feel free to also remove other PR/branche related to packages update for cleaning purpose.

If everything is ok, I can make a new commit to bump the version.
Could you release a new version following this PR? (This would be much appreciated heh)

Have a good day.

@mzuccaroli
